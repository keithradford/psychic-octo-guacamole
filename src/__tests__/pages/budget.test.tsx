import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import BudgetPage, { BudgetSectionTitle } from "../../pages/budget";

jest.mock("../../utils/trpc", () => ({
  trpc: {
    getBudgetByTerm: {
      useQuery: () => ({
        data: {
          budget: {
            budgetBars: [
              {
                id: 1,
                name: "Income",
                max: 1000,
                currentVal: 671,
                isIncome: true,
              },
              {
                id: 2,
                name: "Spend",
                max: 1000,

                currentVal: 671,
                isIncome: false,
              },
            ],
          },
        },
      }),
    },
    addBudgetBarToBudget: {
      useMutation: () => ({
        mutateAsync: () => {},
      }),
    },
    updateBudgetBar: {
      useMutation: () => ({
        mutateAsync: () => {},
      }),
    },
  },
}));

describe("Budget page", () => {
  it("should render correctly", () => {
    const { container } = render(<BudgetPage />);
    expect(container).toMatchSnapshot();
  });
});

describe("Budget section title", () => {
  it("should display the title", () => {
    const { getByText } = render(
      <BudgetSectionTitle title="Income" max={1000} current={671} />
    );
    expect(getByText("Income")).toBeInTheDocument();
  });

  it("should display the current amount", () => {
    const { getByText } = render(
      <BudgetSectionTitle title="Income" max={1000} current={671} />
    );
    expect(getByText(/\$671/)).toBeInTheDocument();
  });

  it("should display the max amount", () => {
    const { getByText } = render(
      <BudgetSectionTitle title="Income" max={1000} current={671} />
    );
    expect(getByText(/\$1000/)).toBeInTheDocument();
  });
});
