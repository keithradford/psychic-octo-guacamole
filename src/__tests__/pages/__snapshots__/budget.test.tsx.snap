// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget page should render correctly 1`] = `
<div>
  <div
    class="flex flex-col w-full space-y-5"
  >
    <div
      class="flex flex-row flex-grow space-x-5"
    >
      <div
        class="flex flex-col space-y-1 w-60"
      >
        <div
          class="flex justify-between"
        >
          <h1
            class="text-xl"
          >
            Income
          </h1>
          <button
            class="px-3 py-1 rounded-lg border border-white bg-primary text-white transition-colors hover:bg-opacity-90 active:bg-opacity-100"
          >
            <svg
              aria-hidden="true"
              class="w-3 h-3"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5v15m7.5-7.5h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
        <p
          class="font-light"
        >
          $
          671
           of $
          1000
        </p>
      </div>
      <div
        class="flex flex-col w-full space-y-5"
      >
        <div
          class="flex flex-col w-full space-y-2 hover:cursor-pointer"
        >
          <div
            class="flex items-end justify-between"
          >
            <h1
              class="text-lg"
            />
            <h2
              class="text-sm font-light"
            >
              $
              671
               of $
              1000
            </h2>
          </div>
          <div
            class="w-full h-4 border border-gray-400 rounded-md"
          >
            <div
              class="bg-green-300 h-full rounded-md"
              data-testid="percent-bar"
              style="width: 67.10000000000001%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full border"
    />
    <div
      class="flex flex-row flex-grow space-x-5"
    >
      <div
        class="flex flex-col space-y-1 w-60"
      >
        <div
          class="flex justify-between"
        >
          <h1
            class="text-xl"
          >
            Spending
          </h1>
          <button
            class="px-3 py-1 rounded-lg border border-white bg-primary text-white transition-colors hover:bg-opacity-90 active:bg-opacity-100"
          >
            <svg
              aria-hidden="true"
              class="w-3 h-3"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5v15m7.5-7.5h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
        <p
          class="font-light"
        >
          $
          671
           of $
          1000
        </p>
      </div>
      <div
        class="flex flex-col w-full space-y-5"
      >
        <div
          class="flex flex-col w-full space-y-2 hover:cursor-pointer"
        >
          <div
            class="flex items-end justify-between"
          >
            <h1
              class="text-lg"
            />
            <h2
              class="text-sm font-light"
            >
              $
              671
               of $
              1000
            </h2>
          </div>
          <div
            class="w-full h-4 border border-gray-400 rounded-md"
          >
            <div
              class="bg-green-300 h-full rounded-md"
              data-testid="percent-bar"
              style="width: 67.10000000000001%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
